lib/cocoapods/installer/analyzer/locking_dependency_analyzer.rb

```diff
@@ -17,7 +17,8 @@ module LockingDependencyAnalyzer
         #         generated by the lockfile that prevent the resolver to update
         #         a Pod.
         #
-        def self.generate_version_locking_dependencies(lockfile, pods_to_update)
+        def self.generate_version_locking_dependencies(lockfile, pods_to_update, pods_to_unlock = nil)
+          @pods_to_unlock = pods_to_unlock
```

Comment:
It means passing through both `add_to_dependency_graph` and `add_child_vertex_to_graph` methods. But it's not a big problem because we currently passing `dependency_graph` as a parameter, so it would be consistent. Will switch to that.